### PR TITLE
Ansible AWS cloud build and deployment

### PR DESCRIPTION
**Pull request by edevenport**
This PR contains the cloud build playbook and AWS roles. The hardcoded apps directory path has also been changed into an Ansible variable.

### CODE REVIEW

File: ansible/roles/build-cloud-infrastructure/tasks/provider/aws.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  until: ec2_keys|succeeded
+  async: 100
+  poll: 0
+
+- name: AWS key
+  debug:
+    msg: "{{ ec2_keys }}"
+
+- set_fact:
+    ingress_rules:
```
Comment (luhkevin) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/aws.yml]: We'll need 2001 (TCP), 2003 (TCP) and 2003 (UDP) as well. More specifically, we'll need those open in the VPC, but not necessarily publicly.

Moreover, is there a way to have these security groups in some common format that can be read by the GCE/Azure/DO/etc... modules as well? Don't know if ansible has a facility for that.

File: ansible/group_vars/cloud.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+cluster_name: "{{ lookup('env', 'CLUSTER_NAME') }}"
+provider: "{{ lookup('env', 'PROVIDER') }}"
+master_size: "{{ lookup('env', 'MASTER_SIZE') }}"
+ssh_private_key: "{{  lookup('env','SSH_PRIVATE_KEY_B64') | b64decode | default( lookup('env','SSH_PRIVATE_KEY'), True) }}"
+ssh_public_key: "{{  lookup('env','SSH_PUBLIC_KEY_B64') | b64decode | default( lookup('env','SSH_PUBLIC_KEY'), True) }}"
+
+### Provider specific variables ###
+region: "{{ lookup('env', 'REGION') }}"
+image: "{{ lookup('env', 'IMAGE') }}"
+root_disk_size: "{{ lookup('env', 'ROOT_DISK_SIZE') or '60' }}"
```
Comment (luhkevin) [inline ansible/group_vars/cloud.yml]: Can we default to 80?

File: ansible/roles/remove-cloud-node/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+- name: Build CoreOS on AWS
+  include: provider/aws.yml
+  when:
+    - provider == 'aws'
+
+- name: Delete node record
```
Comment (luhkevin) [inline ansible/roles/remove-cloud-node/tasks/main.yml]: Is it possible to only run this task if we get a success message from the node removal task in `remove-cloud-node/tasks/provider/aws.yml`?

File: scripts/deploy.sh [hunk 4]
```diff
@@ -12,12 +12,7 @@ fi
 export ANSIBLE_HOST_KEY_CHECKING=0
 export ANSIBLE_SCP_IF_SSH=1
 export ANSIBLE_SUDO_FLAGS="-H -S"
-
-if [ "$DISTRIB" == "coreos" ]; then
-    export ANSIBLE_SSH_ARGS="-o ControlMaster=auto -o ControlPersist=60s -o ControlPath=/opt/ansible.netsil"
-else
-    export ANSIBLE_SSH_ARGS="-o ControlMaster=no"
```
Comment (luhkevin) [inline scripts/deploy.sh]: Hmm, I don't remember the exact issue, but I do think one of or both `-o ControlPersist=60s -o ControlPath=/opt/ansible.netsil` was not working on CentOS, which is why I had to do this conditional.

File: ansible/group_vars/cloud.yml (hunk 2 above)
Comment (edevenport) [inline ansible/group_vars/cloud.yml]: The default size has been increased to 80.

File: ansible/roles/remove-cloud-node/tasks/main.yml (hunk 3 above)
Comment (edevenport) [inline ansible/roles/remove-cloud-node/tasks/main.yml]: Good catch, this has been updated.

File: scripts/deploy.sh (hunk 4 above)
Comment (edevenport) [inline scripts/deploy.sh]: This negates the need for the conditional. I now set the path to `/dev/ansible.netsil` which will work for CoreOS and CentOS. It should work with other distributions as well.

File: ansible/roles/build-cloud-infrastructure/tasks/provider/aws.yml (hunk 1 above)
Comment (edevenport) [inline ansible/roles/build-cloud-infrastructure/tasks/provider/aws.yml]: I added the specified ports to the EC2 security group. This should allow access within the VPC. We are not modifying the network ACL assigned to the existing subnet, so public access will not be permitted unless the network ACL is updated accordingly.

There really isn't a common format for the firewall rules. The cloud modules that I have worked with each have very different formats for handling firewall rules.

**Pull request merged**